### ISSUE TITLE
iOS 12.2 InvalidStateError: Failed to execute 'transaction' on 'IDBDatabase'

### ISSUE THREAD

**Issue by geoffliddiard** [ionic-team/cordova-plugin-ionic-webview#352]
We are seeing the following errors with the Webview in iOS 12.2:

> InvalidStateError: Failed to execute 'transaction' on 'IDBDatabase': The database connection is closing. transaction@[native code] http://localhost:8080/var/containers/Bundle/Application/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/X

and

> DOMException: Failed to execute 'transaction' on 'IDBDatabase': The database connection is closing.

We haven't managed to nail down exact steps to reproduce, but it seems to happen most consistently after the app has been resumed from a suspended/background state after some indeterminate amount of time - the longer the time, the greater the odds of reproducing. So far we have also only been able to reproduce this issue on production (signed) builds, so debugging has been difficult.

We have tried v4.0.1 and v2.4.1 hoping this may have been fixed along with some of the other iOS 12.2 related patches, but no such luck.

My environment:
```
Ionic:

   ionic (Ionic CLI)  : 4.1.2 (/usr/local/lib/node_modules/ionic)
   Ionic Framework    : ionic-angular 3.9.5
   @ionic/app-scripts : 3.2.3

Cordova:

   cordova (Cordova CLI) : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms     : android 8.0.0, ios 5.0.0
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 2.4.1, (and 15 other plugins)

System:

   Android SDK Tools : 26.1.1
   ios-deploy        : 2.0.0
   ios-sim           : 7.0.0
   NodeJS            : v11.9.0 
   npm               : 6.5.0
   OS                : macOS
   Xcode             : Xcode 10.2 Build version 10E125
```

**Comment by geoffliddiard** [comment]
possibly related to this: https://bugs.webkit.org/show_bug.cgi?id=197050

**Comment by geoffliddiard** [comment]
Due to how our app implements support for offline use with indexeddb storage and the volume of our users this has been affecting, we have been forced to release a workaround for this issue. We chose to migrate our IndexedDb storage to SQLite using the [cordova-sqlite-storage](https://github.com/xpbrew/cordova-sqlite-storage) plugin.

**Comment by connor-odoherty** [comment]
We are experiencing the exact same issue as well. 

```
InvalidStateError: Failed to execute 'transaction' on 'IDBDatabase'
DOMException: Failed to execute 'transaction' on 'IDBDatabase': The database connection is closing.
```

Ionic info:
```
Ionic:

   ionic (Ionic CLI)  : 4.12.0
   Ionic Framework    : ionic-angular 3.9.2
   @ionic/app-scripts : 3.2.3

Cordova:

   cordova (Cordova CLI) : 9.0.0
   Cordova Platforms     : ios 5.0.1
   Cordova Plugins       : cordova-plugin-ionic 5.3.0, cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 4.0.1, (and 21 other plugins)

System:

   ios-deploy : 1.9.4
   ios-sim    : 8.0.1
   NodeJS     : v11.14.0 (/usr/local/Cellar/node/11.14.0/bin/node)
   npm        : 6.9.0
   OS         : macOS Mojave
   Xcode      : Xcode 10.2 Build version 10E125

```

Package:
```
{
  "name": "XXX",
  "version": "XXX",
  "author": "XXX",
  "homepage": "XXX/",
  "private": true,
  "scripts": {

  },
  "dependencies": {
    "@angular/common": "5.2.11",
    "@angular/compiler": "5.2.11",
    "@angular/compiler-cli": "5.2.11",
    "@angular/core": "5.2.11",
    "@angular/flex-layout": "^7.0.0-beta.23",
    "@angular/forms": "5.2.11",
    "@angular/http": "5.2.11",
    "@angular/platform-browser": "5.2.11",
    "@angular/platform-browser-dynamic": "5.2.11",
    "@ionic-native/app-version": "4.7.0",
    "@ionic-native/background-mode": "^4.20.0",
    "@ionic-native/badge": "^4.20.0",
    "@ionic-native/base64": "4.7.0",
    "@ionic-native/camera": "^4.7.0",
    "@ionic-native/core": "^4.9.2",
    "@ionic-native/device": "^4.20.0",
    "@ionic-native/document-viewer": "4.7.0",
    "@ionic-native/file": "4.9.2",
    "@ionic-native/file-opener": "4.9.2",
    "@ionic-native/file-transfer": "4.9.2",
    "@ionic-native/in-app-browser": "^4.11.0",
    "@ionic-native/ionic-webview": "^5.0.0-beta.14",
    "@ionic-native/keyboard": "4.7.0",
    "@ionic-native/network": "^4.11.0",
    "@ionic-native/pro": "^4.9.2",
    "@ionic-native/push": "^4.19.0",
    "@ionic-native/safari-view-controller": "^4.11.0",
    "@ionic-native/secure-storage": "^4.7.0",
    "@ionic-native/splash-screen": "4.7.0",
    "@ionic-native/status-bar": "4.7.0", 
    "@ionic-native/themeable-browser": "^4.16.0",
    "@ionic-native/unique-device-id": "4.7.0",
    "@ionic/pro": "2.0.3",
    "@ionic/storage": "2.1.3",
    "body-parser": "^1.18.2",
    "com-badrit-base64": "^0.2.0",
    "connect-wwwhisper": "*",
    "cordova-browser": "5.0.3",
    "cordova-ios": "5.0.1",
    "cordova-plugin-add-swift-support": "^1.7.0",
    "cordova-plugin-app-version": "0.1.9",
    "cordova-plugin-background-mode": "git+https://github.com/sercatven/cordova-plugin-background-mode.git",
    "cordova-plugin-badge": "^0.8.8",
    "cordova-plugin-camera": "^2.4.1",
    "cordova-plugin-compat": "^1.2.0",
    "cordova-plugin-device": "2.0.2",
    "cordova-plugin-device-name": "^1.3.2",
    "cordova-plugin-document-viewer": "^0.9.9",
    "cordova-plugin-file": "6.0.1",
    "cordova-plugin-file-opener2": "^2.0.19",
    "cordova-plugin-file-transfer": "^1.7.1",
    "cordova-plugin-inappbrowser": "^3.0.0",
    "cordova-plugin-ionic": "5.3.0",
    "cordova-plugin-ionic-keyboard": "~2.1.3",
    "cordova-plugin-ionic-webview": "4.0.1",
    "cordova-plugin-network-information": "^2.0.1",
    "cordova-plugin-rollbar": "git+https://github.com/emilyemorehouse/cordova-plugin-rollbar.git",
    "cordova-plugin-safariviewcontroller": "^1.5.4",
    "cordova-plugin-secure-storage": "3.0.1",
    "cordova-plugin-splashscreen": "5.0.2",
    "cordova-plugin-statusbar": "^2.4.2",
    "cordova-plugin-themeablebrowser": "^0.2.17",
    "cordova-plugin-whitelist": "^1.3.3",
    "cors": "^2.8.4",
    "express": "^4.16.2",
    "express-basic-auth": "^1.1.4",
    "heroku-ssl-redirect": "0.0.4",
    "ionic": "4.12.0",
    "ionic-angular": "^3.9.2",
    "ionic2-auto-complete": "^1.6.2-alpha",
    "ionic2-super-tabs": "^4.1.4",
    "ionicons": "3.0.0",
    "json-loader": "^0.5.7",
    "jssha": "^2.3.1",
    "lodash": "^4.17.4",
    "logrocket": "^0.6.11",
    "moment": "^2.21.0",
    "moment-timezone": "^0.5.14",
    "morgan": "^1.9.0",
    "node-sass": "^4.9.2",
    "phonegap-plugin-push": "^2.2.3",
    "reflect-metadata": "^0.1.10",
    "rollbar": "^2.5.1",
    "rxjs": "5.4.0",
    "suncalc": "^1.8.0",
    "sw-toolbox": "3.6.0",
    "typescript": "2.4.0",
    "uglify-js": "^3.3.13",
    "urbanairship-cordova": "7.5.4",
    "webpack-merge": "^4.1.0",
    "zone.js": "0.8.18"
  },
  "devDependencies": {
    "@ionic/app-scripts": "^3.2.3",
    "@types/jssha": "0.0.29",
    "@types/moment": "^2.13.0",
    "@types/moment-timezone": "^0.5.4",
    "tslint": "^5.7.0",
    "ws": "3.3.2"
  },
  "description": "An Ionic project",
  "cordova": {
    "plugins": {
      "cordova-plugin-splashscreen": {},
      "cordova-plugin-whitelist": {},
      "cordova-plugin-document-viewer": {},
      "com-badrit-base64": {},
      "cordova-plugin-device-name": {},
      "cordova-plugin-file": {},
      "cordova-plugin-file-transfer": {},
      "cordova-plugin-file-opener2": {},
      "cordova-plugin-inappbrowser": {},
      "cordova-plugin-safariviewcontroller": {},
      "cordova-plugin-network-information": {},
      "cordova-plugin-ionic-keyboard": {},
      "cordova-plugin-themeablebrowser": {},
      "cordova-plugin-badge": {},
      "cordova-plugin-rollbar": {

      },
      "phonegap-plugin-push": {},
      "cordova-plugin-ionic-webview": {},
      "cordova-plugin-statusbar": {},
      "cordova-plugin-ionic": {
      },
      "cordova-plugin-secure-storage": {},
      "cordova-plugin-app-version": {},
      "cordova-plugin-device": {},
      "cordova-plugin-background-mode": {},
      "urbanairship-cordova": {}
    },
    "platforms": [
      "browser",
      "ios"
    ]
  },
  "config": {
    "ionic_source_map_type": "eval"
  }
}

```

**Comment by ia64mail** [comment]
We see the same issue with iOS 12.2 only. 

```
Ionic:

   ionic (Ionic CLI)  : 4.1.1 (C:\Users\ia64\AppData\Roaming\npm\node_modules\ionic)
   Ionic Framework    : ionic-angular 3.9.2
   @ionic/app-scripts : 3.1.9

Cordova:

   cordova (Cordova CLI) : 7.1.0
   Cordova Platforms     : android 6.3.0
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.0.5, cordova-plugin-ionic-webview 1.2.1, (and 15 other plugins)

System:

   Android SDK Tools : 26.1.1 (C:\Users\xxx\android-sdk)
   NodeJS            : v8.9.4 (C:\Program Files\nodejs\node.exe)
   npm               : 5.6.0
   OS                : Windows 10

```
and `"@ionic/storage": "2.1.3"`

**Comment by jcesarmobile** [comment]
Looks like this was an Apple bug and seems to be fixed in 12.3, can you verify?
